BAZProject/presentation/viewmodels/MovieViewModel.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  MovieViewModel.swift
+//  BAZProject
+//
+//  Created by Ivan Tecpanecatl Martinez on 16/04/23.
+//
+
+import Foundation
+
+class MovieViewModel {
+    
+    let movie: Movie
+    
+    var characters: [Actor] = []
+    var onCharactersUpdate: (() -> Void)?
+    
+    var reviews: [Review] = []
+    var onReviewsUpdate: (() -> Void)?
+    
+    private let repository: MovieRepository
+    
+    init(movie: Movie, repository: MovieRepository) {
+        self.movie = movie
+        self.repository = repository
+    }
+    
+    func fetchCharacters() {
+        repository.getCast(movieID: movie.id) { characters in
```

Comment:
Posible referencia cruzada.
```suggestion
        repository.getCast(movieID: movie.id) { [weak self] characters in
```